src/hooks/useToggle.ts

```diff
@@ -0,0 +1,22 @@
+import { useState } from "react";
+
+
+const useToggle =(defaultValue:boolean)=>{
+
+
+     const [toggledElement, setToggledElement] = useState<boolean>(defaultValue);  
+
+
+    const toggleFunction=()=>{
+        
+        setToggledElement(!toggledElement)        
+   
+
+    }
+
+
+    return {toggleFunction , toggledElement}
```

Comment:
We can return an object or an array. Returning an array will be similar to the useState hook. This will enable easier user of the hook in the same place instead of needing to destructure the object and rename the object items.
```
// with array destructuring
const [password,togglePassword] = useToggle(false);
const [confirmPasswor, toggleConfirmPassword] = useToggle(false)

// with object destructuring
const {value:passowrd, toggleValue:togglePassword} = useToggle(false);
```
so in our case, it will be easier to return an array. Returning an object will be good, if we will return a lot of items and we will just use a small portion of them, so this will make it easier to destructure the items that we need with ctrl+space.

Reply:
done